catalog/rest/rest.go

```diff
@@ -989,3 +989,54 @@ func (r *Catalog) CheckTableExists(ctx context.Context, identifier table.Identif
 	}
 	return true, nil
 }
+
+func (r *Catalog) ListViews(ctx context.Context, namespace table.Identifier) ([]table.Identifier, error) {
+	if err := checkValidNamespace(namespace); err != nil {
+		return nil, err
+	}
+
+	ns := strings.Join(namespace, namespaceSeparator)
+	path := []string{"namespaces", ns, "views"}
+
+	type resp struct {
+		Identifiers []identifier `json:"identifiers"`
+	}
```

Comment:
Looks like you're missing the `next-page-token` here: https://github.com/apache/iceberg/blob/3e6da2e5437ffb3f643275927e5580cb9620256b/open-api/rest-catalog-open-api.yaml#L3882

Probably means we'd also need to implement handling the next-page-token and page-size parameters for use with `doGet` repeatedly to ensure we get ALL the views (or otherwise change this API to enable paging by the caller)

Reply:
@zeroshade Thanks for the quick review. Nice catch. I did thought about it but then I referred to the `ListTables` method that was done previously and realise next page token is not implemented as well.
Do you want to implement it in a separate PR and we fix it for the `ListTables` method as well?

Reply:
How in-depth / complex would adding it be in this PR? 

This PR is currently pretty small so far so as long as adding the handling of `next-page-token` isn't a large amount of code, it's probably better to implement it here and fix `ListTables` rather than do it as a separate PR. I don't think it would be a very large or complex change to add here, but could be wrong.

If it ends up being pretty complex and large to add it here then I'm fine having it implemented in a separate PR.

Give it a try and let me know if it starts spiraling into a significantly large change.

Reply:
Let me give it a try and let you know. I will try to fix both this method and `ListTables` too

Reply:
@zeroshade Could you help to take a look again. I was thinking about leaving the implementation of pagination to the application itself. The method just takes in page-size and page-token and return the list of views and the next token

Reply:
While I like the idea of the application managing it, I don't like having to pass the token + size. It might make more sense to instead have the method return an `iter.Seq2[Identifier, error]` or `iter.Seq2[[]Identifier, error]` and have a default page size that is configurable via catalog properties / catalog config / context value.

That way the interface is still simple (`for err, id := range cat.ListViews(ctx, namespace)`) while still allowing the application to control the iteration.

What do you think?

Reply:
I see. I think its a relatively new feature. Let me try it out. Also, I have been searching for default page size on iceberg and pyiceberg as reference but have not found any. Is it okay to use an arbitrary number of 20?

Reply:
That's fine with me.

@kevinjqliu @Fokko Do you know if pyiceberg supports the paginating that `ListViews` does?

Reply:
i dont think pyiceberg's rest client paginates at all 
https://github.com/apache/iceberg-python/blob/8fcdc951f0959ee2d7e5b9d3057f4b07afd7cdcf/pyiceberg/catalog/rest.py#L737-L746